x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -25,37 +40,29 @@ const checkValueAgainstComparatorMap: {
   [Comparator.LT_OR_EQ]: (a: number, b: number) => a <= b,
 };
 
-export const createLogThresholdExecutor = (alertUUID: string, libs: InfraBackendLibs) =>
+export const createLogThresholdExecutor = (alertId: string, libs: InfraBackendLibs) =>
   async function ({ services, params }: AlertExecutorOptions) {
-    const { count, criteria } = params as LogDocumentCountAlertParams;
     const { alertInstanceFactory, savedObjectsClient, callCluster } = services;
     const { sources } = libs;
+    const castParams = params as LogDocumentCountAlertParams;
+    const { groupBy } = castParams;
 
     const sourceConfiguration = await sources.getSourceConfiguration(savedObjectsClient, 'default');
     const indexPattern = sourceConfiguration.configuration.logAlias;
-
-    const alertInstance = alertInstanceFactory(alertUUID);
+    const alertInstance = alertInstanceFactory(alertId);
 
     try {
-      const query = getESQuery(
-        params as LogDocumentCountAlertParams,
-        sourceConfiguration.configuration
-      );
-      const result = await getResults(query, indexPattern, callCluster);
-
-      if (checkValueAgainstComparatorMap[count.comparator](result.count, count.value)) {
-        alertInstance.scheduleActions(FIRED_ACTIONS.id, {
-          matchingDocuments: result.count,
-          conditions: createConditionsMessage(criteria),
-        });
-
-        alertInstance.replaceState({
-          alertState: AlertStates.ALERT,
-        });
+      const query = getESQuery(castParams, sourceConfiguration.configuration);
+
+      const results =
+        groupBy && groupBy.length > 0
+          ? await getGroupedResults(query, indexPattern, callCluster, GROUP_BY_KEY)
+          : await getUngroupedResults(query, indexPattern, callCluster);
+
+      if (groupBy && groupBy.length > 0) {
+        processGroupByResults({ results, params: castParams, alertInstanceFactory, alertId });
```

Comment:
Good job splitting this up :+1: What do you think about reducing this to a single conditional like

```typescript
if (groupBy && groupBy.length > 0) {
  processGroupedResults(getGroupedResults(...), ...);
} else {
  processUngroupedResults(getUngroupedResults(...), ...);
}
```